R/python.R

```diff
@@ -817,6 +817,94 @@ py_list_attributes <- function(x) {
 }
 
 
+
+#' Get an item from a Python object
+#'
+#' Retrieve an item from a Python object, similar to how
+#' \code{x[name]} might be used in Python code to access an
+#' item called `name` on an object `x`. The object's
+#' `__getitem__` method will be called.
+#'
+#' @param x A Python object.
+#' @param name The item name.
+#' @param silent Boolean; when \code{TRUE}, attempts to access
+#'   missing items will return \code{NULL} rather than
+#'   throw an error.
+#'
+#' @family item-related APIs
+#' @export
+py_get_item <- function(x, name, silent = FALSE) {
```

Comment:
Do these methods need to respect the `convert` attribute of `x`? (i.e. return Python objects rather than R objects when `convert = FALSE`)

Reply:
Good question. Right now, as implemented, we return an R object when the `convert` flag is set, and a Python object otherwise. Do you think we should instead always return a Python object (as `py_get_attr()` does), and instead propagate the `convert` flag?